src/EditorFeatures/Core/Implementation/Formatting/FormatCommandHandler.FormatDocument.cs

```diff
@@ -15,29 +15,29 @@
 using Microsoft.CodeAnalysis.Text;
 using Microsoft.VisualStudio.Commanding;
 using Microsoft.VisualStudio.Text.Editor.Commanding.Commands;
-using Microsoft.VisualStudio.Utilities;
 using Roslyn.Utilities;
 using VSCommanding = Microsoft.VisualStudio.Commanding;
 
 namespace Microsoft.CodeAnalysis.Editor.Implementation.Formatting
 {
     internal partial class FormatCommandHandler : ForegroundThreadAffinitizedObject
     {
+        private const string s_experimentName = "CleanupOn";
+
         public VSCommanding.CommandState GetCommandState(FormatDocumentCommandArgs args)
         {
             return GetCommandState(args.SubjectBuffer);
         }
 
-        private void ShowGoldBarForCodeCleanupConfigurationIfNeeded(Document document)
+        private void ShowGoldBarForCodeCleanupConfiguration(Document document, bool performAdditionalCodeCleanupDuringFormatting)
         {
             AssertIsForeground();
             Logger.Log(FunctionId.CodeCleanupInfobar_BarDisplayed, KeyValueLogMessage.NoProperty);
 
             var workspace = document.Project.Solution.Workspace;
 
-            // if info bar was shown before, no need to show it again
-            if (workspace.Options.GetOption(CodeCleanupOptions.NeverShowCodeCleanupInfoBarAgain, document.Project.Language) ||
-                workspace.Options.GetOption(CodeCleanupOptions.CodeCleanupInfoBarShown, document.Project.Language))
+            // if info bar was shown already in same VS session, no need to show it again
```

Comment:
Can they be moved from the A group to the B group within the same session? @kuhlenh?

Reply:
(my concern being this is another case where they might not see the gold bar if we forced it on them)

Reply:
I don't believe AB test require such precision. it uses a random user's random solution's random behavior. unless those flipping number of users are large enough to influence statistics. we can leave things in simpler form.